ts/src/toobit.ts

```diff
@@ -0,0 +1,2894 @@
+            'quantity': this.currencyToPrecision (currency['code'], amount),
+            'network': networkCode,
+        };
+        if (tag !== undefined) {
+            request['addressExt'] = tag;
+        }
+        const response = await this.privatePostApiV1AccountWithdraw (this.extend (request, params));
+        //
+        // {
+        //     "status": 0,
+        //     "success": true,
+        //     "needBrokerAudit": false, // Do you need a brokerage review?
+        //     "id": "423885103582776064", // Withdrawal successful order id
+        //     "refuseReason":"" // failure rejection reason
+        // }
+        //
+        return this.parseTransaction (response, currency);
+    }
+
+    /**
+     * @method
+     * @name toobit#setMarginMode
+     * @description set margin mode to 'cross' or 'isolated'
+     * @see https://toobit-docs.github.io/apidocs/usdt_swap/v1/en/#change-margin-type-trade
+     * @param {string} marginMode 'cross' or 'isolated'
+     * @param {string} symbol unified market symbol
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} response from the exchange
+     */
+    async setMarginMode (marginMode: string, symbol: Str = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' setMarginMode() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        if (market['type'] !== 'swap') {
+            throw new BadSymbol (this.id + ' setMarginMode() supports swap contracts only');
+        }
+        marginMode = marginMode.toUpperCase ();
+        const request: Dict = {
+            'symbol': market['id'],
+            'marginType': marginMode,
+        };
+        const response = await this.privatePostApiV1FuturesMarginType (this.extend (request, params));
+        //
+        // {"code":200,"symbolId":"BTC-SWAP-USDT","marginType":"ISOLATED"}
+        //
+        return response;
+    }
+
+    /**
+     * @method
+     * @name toobit#setLeverage
+     * @description set the level of leverage for a market
+     * @see https://toobit-docs.github.io/apidocs/usdt_swap/v1/en/#change-initial-leverage-trade
+     * @param {float} leverage the rate of leverage
+     * @param {string} symbol unified market symbol
+     * @param {object} [params] extra parameters specific to the exchange API endpoint
+     * @returns {object} response from the exchange
+     */
+    async setLeverage (leverage: int, symbol: Str = undefined, params = {}) {
+        if (symbol === undefined) {
+            throw new ArgumentsRequired (this.id + ' setLeverage() requires a symbol argument');
+        }
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request: Dict = {
+            'symbol': market['id'],
+            'leverage': this.numberToString (leverage),
```

Comment:
do we need to parse it as an int here? 
<img width="441" height="308" alt="image" src="https://github.com/user-attachments/assets/ab860f71-d971-4b91-b092-1411e66e85c5" />